dyn Write support removed?

I've used to pass `&mut dyn Write` to the templates, but it does not compile any more due to `<W>` bound on template functions lacking `?Sized`.

I think #107 also inhibits calling template functions on the `out` argument of `ToHtml`. To work around #78 I created this code:

```rust
pub struct ResponsiveImage<'asset> {
    asset: &'asset Asset,
    size: Resize,
}

impl<'asset> ResponsiveImage<'asset> {
    pub fn new(asset: &'asset Asset, size: Resize) -> Self {
        ResponsiveImage { asset, size }
    }
}

impl ToHtml for ResponsiveImage<'_> {
    fn to_html(&self, out: &mut dyn Write) -> io::Result<()> {
        let non_retina = self.asset.thumbnail_url(self.size);
        let retina = self.asset.thumbnail_url(self.size * 2);
        templates::responsive_image(out, non_retina, retina)
    }
}
```

Originally I was manually making `write!` calls in `to_html` but the code was a bit messy and I had to consider HTML escaping manually. So I thought I'd create a template and call the template from the impl. It fails to compile though:

```
error[E0277]: the size for values of type `dyn std::io::Write` cannot be known at compilation time
  --> src/utils.rs:71:37
   |
71 |         templates::responsive_image(out, non_retina, retina, "")
   |         --------------------------- ^^^ doesn't have a size known at compile-time
   |         |
   |         required by a bound introduced by this call
   |
   = help: the trait `Sized` is not implemented for `dyn std::io::Write`
note: required by a bound in `responsive_image_html`
  --> /home/wmoore/Projects/bellbird/target/debug/build/bellbird-73ced4a7b3ebeabd/out/templates/template_responsive_image_html.rs:8:30
   |
8  | pub fn responsive_image_html<W>(_ructe_out_: &mut W, non_retina: Origin, retina: Origin, alt: &str) -> io::Result<()>
   |                              ^ required by this bound in `responsive_image_html`
help: consider relaxing the implicit `Sized` restriction
  --> /home/wmoore/Projects/bellbird/target/debug/build/bellbird-73ced4a7b3ebeabd/out/templates/template_responsive_image_html.rs:9:15
   |
9  | where W: Write + ?Sized {
   |                ++++++++
```

--- a/src/template.rs
+++ b/src/template.rs
@@ -41,7 +41,7 @@ impl Template {
         writeln!(
             out,
             "\n\
-             pub fn {name}<{ta}{ta_sep}W>(_ructe_out_: &mut W{args}) -> io::Result<()>\n\
+             pub fn {name}<{ta}{ta_sep}W>(#[allow(unused_mut)] mut _ructe_out_: W{args}) -> io::Result<()>\n\
              where W: Write {{\n\
              {body}\
              Ok(())\n\

--- a/src/templateexpression.rs
+++ b/src/templateexpression.rs
@@ -82,7 +82,7 @@ impl TemplateExpression {
                 format!("_ructe_out_.write_all({text:?}.as_bytes())?;\n")
             }
             TemplateExpression::Expression { ref expr } => {
-                format!("{expr}.to_html(_ructe_out_)?;\n")
+                format!("{expr}.to_html(_ructe_out_.by_ref())?;\n")
             }
             TemplateExpression::ForLoop {
                 ref name,
@@ -122,7 +122,7 @@ impl TemplateExpression {
             ),
             TemplateExpression::CallTemplate { ref name, ref args } => {
                 format!(
-                    "{name}(_ructe_out_{})?;\n",
+                    "{name}(_ructe_out_.by_ref(){})?;\n",
                     args.iter().format_with("", |arg, f| f(&format_args!(
                         ", {arg}"
                     ))),

